src/query/phrase_query/phrase_scorer.rs

```diff
@@ -244,242 +171,32 @@ fn intersection_exists_with_slop(left: &[u32], right: &[u32], slop: u32) -> bool
     false
 }
 
-impl<TPostings: Postings> PhraseScorer<TPostings> {
-    pub fn new(
-        term_postings: Vec<(usize, TPostings)>,
-        similarity_weight: Bm25Weight,
-        fieldnorm_reader: FieldNormReader,
-        scoring_enabled: bool,
-        slop: u32,
-    ) -> PhraseScorer<TPostings> {
-        let max_offset = term_postings
-            .iter()
-            .map(|&(offset, _)| offset)
-            .max()
-            .unwrap_or(0);
-        let num_docsets = term_postings.len();
-        let postings_with_offsets = term_postings
-            .into_iter()
-            .map(|(offset, postings)| {
-                PostingsWithOffset::new(postings, (max_offset - offset) as u32)
-            })
-            .collect::<Vec<_>>();
-        let mut scorer = PhraseScorer {
-            intersection_docset: Intersection::new(postings_with_offsets),
-            num_terms: num_docsets,
-            left: Vec::with_capacity(100),
-            right: Vec::with_capacity(100),
-            phrase_count: 0u32,
-            similarity_weight,
-            fieldnorm_reader,
-            scoring_enabled,
-            slop,
-        };
-        if scorer.doc() != TERMINATED && !scorer.phrase_match() {
-            scorer.advance();
-        }
-        scorer
-    }
-
-    pub fn phrase_count(&self) -> u32 {
-        self.phrase_count
-    }
-
-    fn phrase_match(&mut self) -> bool {
-        if self.scoring_enabled {
-            let count = self.compute_phrase_count();
-            self.phrase_count = count;
-            count > 0u32
-        } else {
-            self.phrase_exists()
-        }
-    }
-
-    fn phrase_exists(&mut self) -> bool {
-        let intersection_len = self.compute_phrase_match();
-        if self.has_slop() {
-            return intersection_exists_with_slop(
-                &self.left[..intersection_len],
-                &self.right[..],
-                self.slop,
-            );
-        }
-        intersection_exists(&self.left[..intersection_len], &self.right[..])
-    }
-
-    fn compute_phrase_count(&mut self) -> u32 {
-        let intersection_len = self.compute_phrase_match();
-        if self.has_slop() {
-            return intersection_count_with_slop(
-                &self.left[..intersection_len],
-                &self.right[..],
-                self.slop,
-            ) as u32;
-        }
-        intersection_count(&self.left[..intersection_len], &self.right[..]) as u32
-    }
-
-    fn compute_phrase_match(&mut self) -> usize {
-        {
-            self.intersection_docset
-                .docset_mut_specialized(0)
-                .positions(&mut self.left);
-        }
-        let mut intersection_len = self.left.len();
-        for i in 1..self.num_terms - 1 {
-            {
-                self.intersection_docset
-                    .docset_mut_specialized(i)
-                    .positions(&mut self.right);
-            }
-            intersection_len = if self.has_slop() {
```

Comment:
We do this to remove these `if self.has_slop` + the unnecessary slop attributes in `ExactPhraseScorer`